runtime/protocol/http-client-engines/http-client-engine-okhttp/jvm/src/aws/smithy/kotlin/runtime/http/engine/okhttp/OkHttpEngine.kt

```diff
@@ -23,12 +22,49 @@ import kotlin.time.toJavaDuration
  * [aws.smithy.kotlin.runtime.http.engine.HttpClientEngine] based on OkHttp.
  */
 class OkHttpEngine(
-    private val config: HttpClientEngineConfig = HttpClientEngineConfig.Default
+    private val config: OkHttpEngineConfig
 ) : HttpClientEngineBase("OkHttp") {
+    public constructor() : this(OkHttpEngineConfig.Default)
 
-    // TODO - expose thread count and use custom dispatcher/ExecutionService
+    public companion object {
+        public operator fun invoke(block: OkHttpEngineConfig.Builder.() -> Unit): OkHttpEngine = OkHttpEngine(
+            OkHttpEngineConfig.Builder().apply(block).build()
+        )
+    }
+
+    private val client = config.buildClient()
+
+    override suspend fun roundTrip(context: ExecutionContext, request: HttpRequest): HttpCall {
```

Comment:
none of this changed, the diff is just confusing. I moved the construction of the client on L54 into a method `buildClient()`